sktime/transformations/base.py

```diff
@@ -134,6 +134,56 @@ def __init__(self):
 
         super(BaseTransformer, self).__init__()
 
+    def __mul__(self, other):
```

Comment:
I am wondering why we need this, ans why is it not `__add__` instead? I dont see a concrete benefit and usecase for using a dunder method. So I would prefer to not implement a dunder method.

Reply:
the benefit is that you can write sth like `my_trafo * my_trafo2` and that´s your pipeline.

Also, "add" seems more natural for `FeatureUnion` rather than for composition.
If you use addition for feature union and multiplication for composition, the pipeline satisfies the familiar distributive law.

Reply:
> I dont see a concrete benefit and usecase for using a dunder method. 

Simple specification of pipelines, increased readibility of pipeline specifications, user comfort.

Reply:
see this for a widely popular design in R which is related
https://cran.r-project.org/web/packages/magrittr/vignettes/magrittr.html